src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAuthenticationProvider.java

```diff
@@ -203,6 +235,26 @@ public Authentication doAuthenticate(Authentication authentication) throws Authe
         }
     }
 
+    private static List<Object> extractAcrAssertion(Response response) {
+        if (response == null) {
+            return null;
+        }
+        Assertion assertion = CollectionUtils.firstElement(response.getAssertions());
+        if (assertion == null) {
+            return null;
+        }
+        List<String> acrValues = assertion
+            .getAuthnStatements()
+            .stream()
+            .map(authStatement -> authStatement.getAuthnContext())
+            .filter(Objects::nonNull)
+            .map(authCtx -> authCtx.getAuthnContextClassRef())
+            .filter(Objects::nonNull)
+            .map(acr -> acr.getURI())
+            .toList();
+        return new ArrayList<Object>(acrValues);
```

Comment:
check if the list is empty and in that case return null. Also avoid creating a new list to copy elements for the result, return the list built by the stream collector

Reply:
>check if the list is empty and in that case return null

Isn't this suggestion in conflict with https://github.com/scc-digitalhub/AAC/pull/504#discussion_r1452233791 ? Am I not understanding the comment correctly?